yt_dlp/extractor/moviepilot.py

```diff
@@ -0,0 +1,119 @@
+            'view_count': int,
+            'uploader_id': 'x6nd9k',
+            'upload_date': '20140519',
+            'duration': 140,
+            'age_limit': 0,
+            'tags': ['Alle Trailer', 'Movie', 'Third Party'],
+        },
+    }, {
+        'url': 'https://www.moviepilot.de/movies/interstellar-2/trailer',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.moviepilot.de/movies/interstellar-2/kinoprogramm/berlin',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.moviepilot.de/movies/queen-slim/trailer',
+        'info_dict': {
+            'id': 'x7xj6o7',
+            'display_id': 'queen-slim',
+            'title': 'Queen & Slim',
+            'ext': 'mp4',
+            'thumbnail': 'https://s2.dmcdn.net/v/SbUM71WtomSjVmI_q/x720',
+            'timestamp': 1571838685,
+            'description': 'md5:73058bcd030aa12d991e4280d65fbebe',
+            'uploader': 'Moviepilot',
+            'like_count': int,
+            'view_count': int,
+            'uploader_id': 'x6nd9k',
+            'upload_date': '20191023',
+            'duration': 138,
+            'age_limit': 0,
+            'tags': ['Movie', 'Verleih', 'Neue Trailer'],
+        },
+    }, {
+        'url': 'https://www.moviepilot.de/movies/der-geiger-von-florenz/trailer',
+        'info_dict': {
+            'id': 'der-geiger-von-florenz',
+            'title': 'Der Geiger von Florenz',
+            'ext': 'mp4',
+        },
+        'skip': 'No trailer for this movie.',
+    }, {
+        'url': 'https://www.moviepilot.de/movies/muellers-buero/',
+        'info_dict': {
+            'id': 'x7xcw1i',
+            'display_id': 'muellers-buero',
+            'title': 'Müllers Büro',
+            'ext': 'mp4',
+            'description': 'md5:57501251c05cdc61ca314b7633e0312e',
+            'timestamp': 1287584475,
+            'age_limit': 0,
+            'duration': 82,
+            'upload_date': '20101020',
+            'thumbnail': 'https://s1.dmcdn.net/v/SaMes1WfAm1d6maq_/x720',
+            'uploader': 'Moviepilot',
+            'like_count': int,
+            'view_count': int,
+            'tags': ['Alle Trailer', 'Movie', 'Verleih'],
+            'uploader_id': 'x6nd9k',
+        },
+    }]
+
+    def _get_property(self, name, webpage, fatal=True):
+        return self._search_regex(f'meta itemprop="{name}" content="(.*?)"', webpage, name, fatal=fatal)
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(f'https://www.moviepilot.de/movies/{video_id}/trailer', video_id)
+
+        # oriented on parse_duration()
+        d = re.match(r'P(?P<hours>[0-9]+)H(?P<mins>[0-9]{2})M(?P<secs>[0-9]{2})S',
+                     self._get_property("duration", webpage, fatal=False))
+        if d:
+            hours, mins, secs = d.groups()
+            duration = float(hours) * 60 * 60 + float(mins) * 60 + float(secs)
+        else:
+            duration = None
```

Comment:
```suggestion
        duration = try_get(
            re.match(r'P(?P<hours>\d+)H(?P<mins>\d+)M(?P<secs>\d+)S',
                     self._html_search_meta('duration', webpage, fatal=False) or ''),
            lambda mobj: sum(float(x) * y for x, y in zip(mobj.groups(), (3600, 60, 1))))
```
(can be split into 2 expressions if u think it's too complicated)

Reply:
Complexity is fine so.

What do you think of?
```diff
- lambda mobj: sum(float(x) * y for x, y in zip(mobj.groups(), (3600, 60, 1))))
+ lambda mobj: float(sum(x * y for x, y in zip(mobj.groups(), (3600, 60, 1))))
```
Is the float conversion necessary at all?

Reply:
it won't work since values returned by regex are strings

Reply:
Ok.